Source/Core/InputCommon/ControllerInterface/WGInput/WGInput.h

```diff
@@ -0,0 +1,222 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#pragma warning(push)
+#pragma warning(disable : 4265)
+#include <wrl.h>
+#pragma warning(pop)
+
+#include <windows.gaming.input.h>
+
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+
+namespace ciface
+{
+namespace WGInput
+{
+namespace WGI = ABI::Windows::Gaming::Input;
+using Microsoft::WRL::ComPtr;
+
+void Init();
+void DeInit();
+void PopulateDevices();
+
+class Device : public Core::Device
+{
+protected:
+  class Button : public Core::Device::Input
+  {
+  public:
+    Button(const boolean& _button) : m_button(_button) {}
```

Comment:
boolean? What language is this? :)

Reply:
constructor should also be marked with `explicit`

Reply:
"boolean" is from the windows API. It's a typedef for unsigned char. :/ I'll add a comment.

Reply:
Also: isn't the first-underscore naming reserved (and thus should be avoided here)?
What's wrong with just calling it `button` since the member is `m_button`?
_(Applies to other places as well)_